sphinx/ext/autodoc/__init__.py

```diff
@@ -1157,10 +1157,7 @@ def format_args(self, **kwargs: Any) -> str:
 
         try:
             self.env.app.emit('autodoc-before-process-signature', self.object, False)
-            if inspect.is_singledispatch_function(self.object):
-                sig = inspect.signature(self.object, follow_wrapped=True)
-            else:
-                sig = inspect.signature(self.object)
+            sig = inspect.signature(self.object, follow_wrapped=True)
```

Comment:
@tk0miya Should we specify `followed_wrapeed=True` in `inspect.signature` in `ClassDocumenter` otherwise a wrapper function's signature would be used when ``__init__`` is decorated?

Reply:
The same goes for:

https://github.com/sphinx-doc/sphinx/blob/38b868cc0d0583d9a58496cd121f0bc345bf9eaa/sphinx/ext/autodoc/type_comment.py#L122